File: gdscript/src/main/java/gdscript/psi/impl/GdAssignTypedImpl.java
package gdscript.psi.impl;

import org.jetbrains.annotations.*;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import com.intellij.extapi.psi.ASTWrapperPsiElement;
import gdscript.psi.*;

public class GdAssignTypedImpl extends ASTWrapperPsiElement implements GdAssignTyped {

  public GdAssignTypedImpl(@NotNull ASTNode node) {
    super(node);
  }

  public void accept(@NotNull GdVisitor visitor) {
    visitor.visitAssignTyped(this);
  }

  @Override
  public void accept(@NotNull PsiElementVisitor visitor) {
    if (visitor instanceof GdVisitor) accept((GdVisitor)visitor);
    else super.accept(visitor);
  }

}

File: gdscript/src/main/kotlin/gdscript/psi/impl/GdAssignTypedImpl.kt
package gdscript.psi.impl

import com.intellij.extapi.psi.ASTWrapperPsiElement
import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElementVisitor
import gdscript.psi.GdAssignTyped
import gdscript.psi.GdVisitor

class GdAssignTypedImpl(node: ASTNode) : ASTWrapperPsiElement(node), GdAssignTyped {
    fun accept(visitor: GdVisitor) {
        visitor.visitAssignTyped(this)
    }

    override fun accept(visitor: PsiElementVisitor) {
        if (visitor is GdVisitor) accept(visitor)
        else super.accept(visitor)
    }
}
